### ISSUE TITLE
Calendar - date should be selectable by pressing 'space'

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7337]
### Expected Behavior
Dates in the Calendar should be selectable by clicking, pressing 'space' or 'enter'

### Actual Behavior
Pressing 'space' doesn't work to select the date

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/visualizations-calendar-simple--simple

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by Erayushsingh** [comment]
@jcfilben Can you please assign this issue to me ? I can work on it .
We may add keyboard event listeners to resolve this issue.

**Comment by britt6612** [comment]
@Erayushsingh we usually dont assign issue however feel free to put up a PR

**Comment by avantika-codes** [comment]
Picking this up
